components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementAdminService.java

```diff
@@ -527,8 +527,8 @@ public PIPFinderDataHolder getPIPResourceFinderData(String finder) {
      */
     public String getGlobalPolicyAlgorithm() throws EntitlementException {
 
-        return EntitlementAdminEngine.getInstance().
-                getPolicyDataStore().getGlobalPolicyAlgorithmName();
+        ConfigDAO config = new RegistryConfigDAOImpl();
```

Comment:
Initializing this should be handled more centrally as mentioned in https://github.com/wso2/carbon-identity-framework/pull/5673/files#r1595074332